server.go

```diff
@@ -653,6 +653,22 @@ func (s *Server) User(name string) *User {
 	return s.users[name]
 }
 
+// AuthenticatedUser returns an authenticated user by username. If any error occurs,
+// or the authentication credentials are invalid, an error is returned.
+func (s *Server) AuthenticatedUser(username, password string) (*User, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	u := s.users[username]
+	if u == nil {
+		return nil, fmt.Errorf("user not found")
+	}
+	err := u.Authenticate(password)
+	if err != nil {
+		return nil, fmt.Errorf("invalid credentials")
+	}
+	return u, nil
+}
```

Comment:
@benbjohnson -- let me explain my thinking here, see if it makes sense. If not, I am happy to change, as I want to do what makes most sense for the *next* guy. 

We have `User()`, which given a username returns a `*User` type. Therefore, a function which also returns `*User` type, but whose auth creds have been checked should be named `AuthenticatedUser()`. Make sense, or is it still counter-intuitive?